Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -380,5 +325,150 @@ public T Deserialize<T>(IIonReader reader)
         {
             return (T) Deserialize(reader, typeof(T));
         }
+        
+        private bool ValidateCustomSerializer(Type type, dynamic serializer)
+        {
+            if (!this.primitiveSerializers.ContainsKey(type))
+            {
+                throw new NotSupportedException($"Custom serializer for {type} is not supported");
+            }
+
+            if (type == typeof(bool))
+            {
+                return serializer is IonSerializer<bool>;
+            }
+
+            if (type == typeof(string))
+            {
+                return serializer is IonSerializer<string>;
+            }
+
+            if (type == typeof(byte[]))
+            {
+                return serializer is IonSerializer<byte[]>;
+            }
+
+            if (type == typeof(int))
+            {
+                return serializer is IonSerializer<int>;
+            }
+
+            if (type == typeof(long))
+            {
+                return serializer is IonSerializer<long>;
+            }
+
+            if (type == typeof(float))
+            {
+                return serializer is IonSerializer<float>;
+            }
+
+            if (type == typeof(double))
+            {
+                return serializer is IonSerializer<double>;
+            }
+
+            if (type == typeof(decimal))
+            {
+                return serializer is IonSerializer<decimal>;
+            }
+
+            if (type == typeof(BigDecimal))
+            {
+                return serializer is IonSerializer<BigDecimal>;
+            }
+
+            if (type == typeof(SymbolToken))
+            {
+                return serializer is IonSerializer<SymbolToken>;
+            }
+
+            if (type == typeof(DateTime))
+            {
+                return serializer is IonSerializer<DateTime>;
+            }
+
+            if (type == typeof(Guid))
+            {
+                return serializer is IonSerializer<Guid>;
+            }
+
+            return false;
+        }
+
+        private void SerializePrimitive(Type type, IIonWriter writer, object item)
+        {
+            var serializer = this.GetPrimitiveSerializer(type);
+
+            if (type == typeof(bool))

```

Comment:
@alpian I'd assume this is hot code, wouldn't we prefer a switch statement for O(1) lookup (it reduces to a lookup table) instead of the fall-through if/else?